pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -211,6 +213,23 @@ func (cs *ConntrackConnectionStore) AddOrUpdateConn(conn *flowexporter.Connectio
 		existingConn.ReverseBytes = conn.ReverseBytes
 		existingConn.ReversePackets = conn.ReversePackets
 		existingConn.TCPState = conn.TCPState
+
+		prevIsActive := existingConn.PriorityQueueItem.IsActive
+		// Reset the pqItem idle expire time and set IsActive to be true when
+		// there are changes either in stats or TCP state.
+		if (conn.OriginalPackets > existingConn.PriorityQueueItem.PrevPackets) ||
+			(conn.ReversePackets > existingConn.PriorityQueueItem.PrevReversePackets) ||
+			(existingConn.PriorityQueueItem.Conn.TCPState != conn.TCPState) {
+			existingConn.PriorityQueueItem.IsActive = true
+			existingConn.PriorityQueueItem.IdleExpireTime = time.Now().Add(cs.ConnectionStore.ExpirePriorityQueue.IdleFlowTimeout)
+		}
+		// We need to update the connection value copy stored in the pqItem.
+		if prevIsActive {
```

Comment:
This is ok. Wondering why doesn't checking pqItem for nil wor here?

Reply:
I didn't quite get that. Are you saying we should do a nil check here to avoid nil pointer error, or we could use nil check to replace isActive flag? conn.pqItem is never marked as nil once it is initialized in the current logic, because we are using it in line 217-224 for every exsiting connection.